frontend/src/metabase/query_builder/components/view/QuestionRowCount/QuestionRowCount.unit.spec.tsx

```diff
@@ -0,0 +1,213 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { fireEvent, render, screen, waitFor } from "__support__/ui";
+
+import type { Dataset } from "metabase-types/api";
+import { createMockDataset } from "metabase-types/api/mocks";
+
+import { HARD_ROW_LIMIT } from "metabase-lib/queries/utils";
+import type Question from "metabase-lib/Question";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import {
+  getSavedStructuredQuestion,
+  getAdHocQuestion,
+  getSavedNativeQuestion,
+  getUnsavedNativeQuestion,
+} from "metabase-lib/mocks";
+
+import QuestionRowCount from "./QuestionRowCount";
+
+type SetupOpts = {
+  question: Question;
+  result?: Dataset;
+  isResultDirty?: boolean;
+};
+
+function setup({
+  result = createMockDataset(),
+  isResultDirty = false,
+  ...props
+}: SetupOpts) {
+  const onQueryChange = jest.fn();
+
+  render(
+    <QuestionRowCount
+      {...props}
+      result={result}
+      isResultDirty={isResultDirty}
+      onQueryChange={onQueryChange}
+    />,
+  );
+
+  const rowCount = screen.getByLabelText("Row count");
+
+  function getNextQuery() {
+    const [lastCall] = onQueryChange.mock.calls.reverse();
```

Comment:
I don't think it's the best practice tho — `reverse` is mutable 🤦 Works well here tho because we're sure we'd only need to access these arguments once. I think we should double-check if we can use the new fancy [`Array.prototype.at` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) that makes it possible to do `calls.at(-1)`

Reply:
TIL: Wow! I've never known `someArray.at(index)` exists. This is really cool.